packages/cli/src/config/parseStoreConfig.ts

```diff
@@ -87,45 +154,33 @@ export interface StoreUserConfig {
    *  - `SchemaType | userType` for a single-value table (aka ECS component).
    *  - FullTableConfig object for multi-value tables (or for customizable options).
    */
-  tables: Record<string, z.input<typeof FieldData> | FullTableConfig>;
+  tables: TablesConfig;
   /** User-defined types that will be generated and may be used in table schemas instead of `SchemaType` */
   userTypes?: UserTypesConfig;
 }
 
-export interface FullTableConfig {
-  /** Output directory path for the file. Default is "/tables" */
-  directory?: string;
-  /**
-   * The fileSelector is used with the namespace to register the table and construct its id.
-   * The table id will be uint256(bytes32(abi.encodePacked(bytes16(namespace), bytes16(fileSelector)))).
-   * Default is "<tableName>"
-   * */
-  fileSelector?: string;
-  /** Make methods accept `tableId` argument instead of it being a hardcoded constant. Default is false */
-  tableIdArgument?: boolean;
-  /** Include methods that accept a manual `IStore` argument. Default is false. */
-  storeArgument?: boolean;
-  /** Include a data struct and methods for it. Default is false for 1-column tables; true for multi-column tables. */
-  dataStruct?: boolean;
-  /** Table's primary key names mapped to their types. Default is `{ key: SchemaType.BYTES32 }` */
-  primaryKeys?: Record<string, z.input<typeof PrimaryKey>>;
-  /** Table's column names mapped to their types. Table name's 1st letter should be lowercase. */
-  schema: Record<string, z.input<typeof FieldData>>;
-}
+export type StoreConfig = z.output<typeof StoreConfig>;
 
-interface UserTypesConfig<Enums extends Record<string, string[]> = Record<string, string[]>> {
-  /** Path to the file where common types will be generated and imported from. Default is "/types" */
-  path?: string;
-  /** Enum names mapped to lists of their member names */
-  enums?: Enums;
-}
+const StoreConfigUnrefined = z.object({
+  namespace: Selector.default(""),
+  storeImportPath: z.string().default("@latticexyz/store/src/"),
+  tables: TablesConfig,
+  userTypes: UserTypesConfig,
+});
 
-export type StoreConfig = z.output<typeof StoreConfig>;
+// finally validate global conditions
+export const StoreConfig = StoreConfigUnrefined.superRefine(validateStoreConfig);
 
-export function parseStoreConfig(config: StoreUserConfig) {
+export function parseStoreConfig(config: unknown) {
```

Comment:
Did this have to get reverted?

Reply:
this was an accidental revert actually - was adapting these changes was from code before your PR and missed this. But good to know all works well